app/discord_message_utils.py

```diff
@@ -0,0 +1,37 @@
+def get_unclosed_code_block_language(message_chunk: str) -> str | None:
+    """\
+    Given an input string representing Discord message content,
+    find the last unclosed code block and return its language.
+
+    For example, for the following text: "```python\nprint('Hello, World!')",
+    the function should return "python".
+
+    NOTE: An empty string is considered as a valid language.
+    """
+    # Even means all blocks were closed correctly.
+    if message_chunk.count("```") % 2 == 0:
```

Comment:
TODO: this is not really true because of `"\\```"

Reply:
```suggestion
    # TODO: this is not really true because of "\```"
    if message_chunk.count("```") % 2 == 0:
```